modules/ximgproc/src/thinning.cpp

```diff
@@ -0,0 +1,67 @@
+#include "precomp.hpp"
+
+using namespace std;
+
+namespace cv {
+namespace ximgproc {
+
+// Function prototypes to suppress -Wmissing-declarations warnings
+void thinningIteration(Mat img, int iter);
+void thinning(InputArray _input, OutputArray _output);
+
+// Applies a thinning iteration to a binary image
+void thinningIteration(Mat img, int iter){
```

Comment:
Okay will do so! Could you explain why the Ni Black hasn't got the ximgproc include but doesnt trigger an issue with the -Wmissing declarations?